src/fparser/two/tests/test_comments_and_directives.py

```diff
@@ -522,12 +528,33 @@ def test_all_directive_formats(directive, expected, free):
     )
     program = Program(reader)
     out = walk(program, Directive)
-    assert len(out) == 1
-    assert out[0].items[0] == expected
+    assert len(out) == len(expected)
+    for i, direc in enumerate(out):
+        assert direc.items[0] == expected[i]
 
     # Test that we correctly get directives without ignore_comments=False.
     reader = get_reader(source, isfree=free, process_directives=True)
     program = Program(reader)
     out = walk(program, Directive)
-    assert len(out) == 1
-    assert out[0].items[0] == expected
+    assert len(out) == len(expected)
+    for i, direc in enumerate(out):
+        assert direc.items[0] == expected[i]
+
+    # Test that we get comments with ignore_comments only?
```

Comment:
I find difficult to follow what the asserts below expect, I would prefer moving it to a differnt test with just a snippet of fortran code with comments and directives on it and then multiple calls to get_reader with different arguments and see that the total number of Comment and Directive match the expectations

Reply:
I simplified this to more understandable and added more comments. I prefer to keep the tests as parametrized as if something breaks its very clear all of the tests that are now failing, otherwise you just stop at the first failure.